src/modules/Hosts/HostsModuleInterface/dllmain.cpp

```diff
@@ -104,11 +110,55 @@ class HostsModuleInterface : public PowertoyModuleIface
     }
 
 public:
+    EventWaiter m_showEventWaiter;
+
+    EventWaiter m_showAdminEventWaiter;
+
     HostsModuleInterface()
     {
         app_name = GET_RESOURCE_STRING(IDS_HOSTS_NAME);
         app_key = ModuleKey;
         LoggerHelpers::init_logger(app_key, L"ModuleInterface", LogSettings::hostsLoggerName);
+
+        m_hShowEvent = CreateDefaultEvent(CommonSharedConstants::SHOW_HOSTS_EVENT);
+        if (!m_hShowEvent)
+        {
+            Logger::error(L"Failed to create show hosts event");
+            auto message = get_last_error_message(GetLastError());
+            if (message.has_value())
+            {
+                Logger::error(message.value());
+            }
+        }
+
+        m_hShowAdminEvent = CreateDefaultEvent(CommonSharedConstants::SHOW_HOSTS_ADMIN_EVENT);
+        if (!m_hShowEvent)
+        {
+            Logger::error(L"Failed to create show hosts admin event");
+            auto message = get_last_error_message(GetLastError());
+            if (message.has_value())
+            {
+                Logger::error(message.value());
+            }
+        }
+
+        m_showEventWaiter = EventWaiter(CommonSharedConstants::SHOW_HOSTS_EVENT, [&](int err)
+        {
+            if (err == ERROR_SUCCESS)
```

Comment:
We need some checks here to verify if the module is enabled before trying to start the process.